pkg/controllers/customroute/custom_route_controller.go

```diff
@@ -0,0 +1,509 @@
+				Reason:             "RouteNotAdmitted",
+				Message:            fmt.Sprintf("Route not admitted: %v", err),
+			},
+			{
+				LastTransitionTime: now,
+				Type:               "Progressing",
+				Status:             metav1.ConditionFalse,
+				Reason:             "RouteNotAdmitted",
+				Message:            fmt.Sprintf("Route not admitted: %v", err),
+			},
+		}
+	}
+
+	if err := c.checkRouteAvailable(route.Spec.Host, ingress); err != nil {
+		return []metav1.Condition{
+			{
+				LastTransitionTime: now,
+				Type:               "Degraded",
+				Status:             metav1.ConditionFalse,
+				Reason:             "ErrorReachingOutToService",
+				Message:            fmt.Sprintf("unexpected error at %s: %v", route.Spec.Host, err),
+			},
+			{
+				LastTransitionTime: now,
+				Type:               "Progressing",
+				Status:             metav1.ConditionTrue,
+				Reason:             "ErrorReachingOutToService",
+				Message:            fmt.Sprintf("unexpected error at %s: %v", route.Spec.Host, err),
+			},
+		}
+	}
+
+	return []metav1.Condition{
+		{
+			LastTransitionTime: now,
+			Type:               "Degraded",
+			Status:             metav1.ConditionFalse,
+			Reason:             "RouteIsAvailable",
+			Message:            "Route successfully configured",
+		},
+		{
+			LastTransitionTime: now,
+			Type:               "Progressing",
+			Status:             metav1.ConditionFalse,
+			Reason:             "RouteIsAvailable",
+			Message:            "Route successfully configured",
+		},
+	}
+}
+
+func (c *customRouteController) checkRouteAvailable(host string, ingress *configv1.Ingress) error {
+	url := "https://" + host
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return err
+	}
+
+	reqCtx, cancel := context.WithTimeout(context.TODO(), 10*time.Second) // avoid waiting forever
+	defer cancel()
+	req.WithContext(reqCtx)
+
+	secret, isDefault, err := common.GetActiveRouterSecret(c.secretLister, "openshift-authentication", "v4-0-config-system-router-certs", "openshift-authentication", "v4-0-config-system-custom-router-certs")
+	if err != nil {
+		return err
+	}
+
+	// Modify HTTP client based on serving certificates in use
+	key := ingress.Spec.Domain
+	if !isDefault {
+		key = corev1.TLSCertKey
+	}
+
+	data, ok := secret.Data[key]
+	if !ok {
+		return fmt.Errorf("secret %s does not contain expected key %s", secret.GetName(), key)
+	}
+
+	rootCAs := x509.NewCertPool()
+	if ok := rootCAs.AppendCertsFromPEM([]byte(data)); !ok {
+		return err
+	}
+
+	c.httpClient.Transport = &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		TLSClientConfig: &tls.Config{
+			RootCAs: rootCAs,
+		},
+	}
+
+	// Make a request to the endpoint, expect a 403
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 403 {
+		return fmt.Errorf("%q returned %d", url, resp.StatusCode)
+	}
+
+	if resp.TLS == nil {
+		return fmt.Errorf("response TLS field is nil")
```

Comment:
this is not a very user friendly message

Reply:
How does `fmt.Errorf("Unable to retrieve TLS information from %s", url)` sound?

Reply:
much better :+1:

Reply:
Updated.